### DISCUSSION TITLE
Parent to child data flow

### DISCUSSION THREAD

**Discussion by logistor** [nuejs/nue#305] [Q&A]
I have following layout.html

```
<main>
  <parent></parent>
</main>
```

and nue template

```
<div @name="parent">
  <child :msg="message"></child>
  <button @click="change">Change</button>
  <script>
  constructor() {
    this.message = "default text"
  }
  change() {
    this.message = "changed text"
  }
  </script>
</div>

<div @name="child">
  <p>{ msg }</p>
</div>
```

I have understood that the data should flow from parent to child.
Can someone tell me what I am doing wrong because I can not get the child text updated when "Change" button is pressed?

**Comment by tipiirai** [comment]
Probably a bug in Nue JS. Unfortunately, the development focus (at least for me) is currently in other parts of the Nue ecosystem so you'll have to wait for some time until this gets fixed.

**Reply by logistor** [reply]
I found one workaround by updating the child manually

```
change() {
    this.message = "changed text"
    const child = this.$refs['child']
    child.update()
  }
```
For some reason this workaround needs <script> tags (even empty ones) in child template to work.

```
<div @name="child">
  <p>{ msg }</p>
  <script>
  </script>
</div>
```

**Reply by tipiirai** [reply]
Good to hear. Thanks!
